candump.c

```diff
@@ -647,16 +751,23 @@ int main(int argc, char **argv)
 				msg.msg_controllen = sizeof(ctrlmsg);  
 				msg.msg_flags = 0;
 
+				idx = idx2dindex(addr.can_ifindex, s[i]);
```

Comment:
Additionally to the fact that I don't like the patch addr.can_ifindex is only valid AFTER recvmsg() ... :-)